### PR TITLE
Design tweaks and functional test for SQL db restore command

### PR DESCRIPTION
**Pull request by jaredmoo**

### CODE REVIEW

**Comment by jaredmoo** [comment]
This is based on #2351 , so after that PR is merged I will rebase this one.

**Comment by jaredmoo** [comment]
Rebase is done.

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py [hunk 1]
```diff
@@ -219,9 +225,21 @@ def _configure_db_create_params(
 with ParametersContext(command='sql db restore') as c:
     _configure_db_create_params(c, Engine.db, CreateMode.point_in_time_restore)
 
-    c.register_alias('requested_service_objective_name', ('--dest-service-objective',))
-    c.register_alias('elastic_pool_name', ('--dest-elastic-pool',))
-    c.register_alias('dest_resource_group_name', ('--dest-resource-group',))
+    c.argument('dest_name',
+               help='Name of the database that will be created as the restore destinaton.')
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py]: typo: `destinaton`

Comment (jaredmoo) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py]: Fixed in 786066f . Good catch!

**Comment by jaredmoo** [comment]
Thanks, that's another one down :)

**Pull request merged**